subprojects/plugins/src/integTest/groovy/org/gradle/api/plugins/ProjectInternalViewDependencyIntegrationTest.groovy

```diff
@@ -0,0 +1,149 @@
+/*
+ * Copyright 2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.api.plugins
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+
+/**
+ * Tests that {@code JvmComponentDependencies#projectInternalView()} behaves as expected.
+ */
+class ProjectInternalViewDependencyIntegrationTest extends AbstractIntegrationSpec {
+
+    def "can test against the internal view of java application"() {
+        given:
+        buildFile << """
+            plugins {
+                id 'application'
+            }
+
+            dependencies {
+                runtimeOnly "com.h2database:h2:2.1.214"
+                implementation "com.fasterxml.jackson.core:jackson-databind:2.13.3"
+                compileOnly "org.apache.commons:commons-lang3:3.12.0"
+            }
+        """
+        writeBaseBuildFile()
+        writeSuccessfulNonApiAccessibilityTest()
+
+        when:
+        succeeds "customTest"
+
+        then:
+        result.assertTaskExecuted(":jar")
+        result.assertTaskExecuted(":customTest")
+    }
+
+    def "can test against the internal view of java library"() {
+        given:
+        buildFile << """
+            plugins {
+                id 'java-library'
+            }
+
+            dependencies {
+                runtimeOnly "com.h2database:h2:2.1.214"
+                implementation "com.fasterxml.jackson.core:jackson-databind:2.13.3"
+                compileOnly "org.apache.commons:commons-lang3:3.12.0"
+
+                api "org.springframework:spring-core:5.3.22"
+                compileOnlyApi "com.google.guava:guava:31.1-jre"
+            }
+        """
+        writeBaseBuildFile()
+        writeSuccessfulNonApiAccessibilityTest()
+        writeSuccessfulApiAccessibilityTest()
+
+        when:
+        succeeds "customTest"
+
+        then:
+        result.assertTaskExecuted(":jar")
+        result.assertTaskExecuted(":customTest")
+    }
+
+    def writeBaseBuildFile() {
```

Comment:
Split this into 2 methods with better names?  I'd have a `setup` method write the build file's `plugins`, `repos` and `testing` blocks, then only do the test specific top-level `dependencies` blocks in each test's `given`.  Then this can be renamed `writeSource()`.  It's moderately nitpicky, but I think it would be clearer - this method's name is a little misleading.

Reply:
The problem with this is that each test uses a different plugin, so their `plugins` block must be different, and therefore the `setup` method can't write to the build-script.